onedal/datatypes/_data_conversion.py

```diff
@@ -18,28 +18,61 @@
 
 import numpy as np
 
-from daal4py.sklearn._utils import make2d
 from onedal import _backend, _is_dpc_backend
 
-from ..utils import _is_csr
-
 
 def _apply_and_pass(func, *args, **kwargs):
     if len(args) == 1:
         return func(args[0], **kwargs)
     return tuple(map(lambda arg: func(arg, **kwargs), args))
 
 
-if _is_dpc_backend:
+def _convert_one_to_table(arg):
+    # All inputs for table conversion must be array-like or sparse, not scalars
+    return _backend.to_table(np.atleast_2d(arg) if np.isscalar(arg) else arg)
+
+
+def to_table(*args):
+    """Create oneDAL tables from scalars and/or arrays.
+
+    Note: this implementation can be used with contiguous scipy.sparse, numpy
+    arrays, sycl_usm_ndarrays, and scalars. Tables will use pointers to the
+    original array data. Scalars will be copies. Arrays may be modified in-
+    place by oneDAL during computation. This works for data located on CPU and
+    SYCL-enabled Intel GPUs. Only singular datatypes are allowed.
```

Comment:
The term `singular datatype` looks strange to me. Is it possible to come up with a more understandable wording? 'numeric data types', 'integer and floating point data types' - something like that.
Or maybe I didn't get what is meant here.

Reply:
Thank you for the suggestion, have modified it to be clearer.